src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,228 @@
+
+import Icon from './Icon';
+
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * Methods to bind in 'this' context
+ */
+const METHODS_TO_BIND = [
+  'toggleExpanded'
+];
+
+/**
+ * An error bar component commonly used in the ServiceForm and the JobsForm
+ * that contains collapsible error messages.
+ *
+ * @example <caption>How to use the CollapsibleErrorMessage</caption>
+ *   <CollapsibleErrorMessage
+ *
+ *        className="error-for-modal"
+ *
+ *        details={[ 'an', 'array', 'of', 'detailed errors' ]}
+ *        message="The message to display"
+ *
+ *        onToggle={function(isToggled) {
+ *          // Do something
+ *        }}
+ *
+ *      />
+ */
+class CollapsibleErrorMessage extends React.Component {
+
+  /**
+   * Initialize superclass
+   */
+  constructor() {
+    super(...arguments);
+
+    // Initial state
+    this.state = {
+      expanded: false
+    };
+
+    // Bind methods in context
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+  }
+
+  /**
+   * Set the default expandedg from the properties
+   */
+  componentWillMount() {
+    this.setState({
+      expanded: this.props.expanded
+    });
+  }
+
+  /**
+   * Toggle the expanded state of the collapsible error bar
+   */
+  toggleExpanded() {
+    let expanded = !this.state.expanded;
+    this.setState({expanded});
+
+    // Call onToggle callback
+    this.props.onToggle(expanded);
+  }
+
+  /**
+   * [Custom component] The (show more/less) link
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getDetailLink() {
+    let {message, details} = this.props;
+    let isDetailed = message && details && details.length;
+
+    // Check if we must not show the detail link at all
+    if (!isDetailed) {
+      return null;
+    }
+
+    // Get text
+    let moreLess = 'more';
+    if (this.state.expanded) {
+      moreLess = 'less';
+    }
+
+    // Render
```

Comment:
Yeah I agree with @rcorral – and I'd actually say comments should be added sparsely and only when they actually add value.
There is a cost in maintaining comments relevant and up to date, and there is nothing worse than an outdated comment that is actually misleading the reader!

For example, the comment `// Render` (while not being a potential liability), adds absolutely zero value, because the `render` function is self-explanatory.

The other comment below, at line 103 instead:
```
/**
* [Fragment] The fixed message part
*
* @returns {React.Component} - The rendered content
*/
getFixedMessagePart() { // ...
```
makes me wonder about two things:
1) it seems to be introducing a very specific "syntax" (`[Fragment]` etc). In this case I'd argue it's introducing a style that seems inconsistent with other components, but maybe this is what the team agrees for the future? In any case I'd suggest tracking the effort to achieve consistent commenting style in the codebase
2) this function may actually return `null` in certain circumstances. The comment does not cater for that case, though.

All in all I'm not saying this should block the PR – but I wanted to share my thoughts on what I think are useful comments and what is actually unnecessary boilerplate. 

On the other hand, this is a good example of a useful comment:

https://github.com/dcos/dcos-ui/blob/master/src/js/utils/ContainerValidatorUtil.js#L7-L9
In this case, the comment is invaluable because it will tell the "future you" where that regex came from and what it means, so you don't have to spend an hour to figure it out. 

Just my 2 cents!